DOCS-1285: Document Micro-RDK supported models (#2072)
viamrobotics/docs: docs/micro-rdk/board/esp32.md

---
title: "Configure an ESP32 board"
linkTitle: "esp32"
weight: 20
type: "docs"
description: "Configure an esp32 board."
images: ["/icons/components/board.svg"]
tags: ["board", "components"]
# SMEs: Gautham, Rand
---

{{% alert title="REQUIREMENTS" color="caution" %}}

> Follow the [setup guide](/installation/prepare/microcontrollers/) to prepare your ESP32 for running micro-RDK before configuring a `esp32` board.

Reviewer: ```suggestion
Follow the [setup guide](/installation/prepare/microcontrollers/) to prepare your ESP32 for running the micro-RDK before configuring an `esp32` board.
```

{{% /alert %}}

To add an `esp32` board, navigate to the **Config** tab of your robot's page in [the Viam app](https://app.viam.com) and select **Raw JSON** mode.

Copy the following JSON template and paste it into your configuration inside the `"components"` array:

{{< tabs name="Configure an esp32 Board" >}}
{{% tab name="JSON Template"%}}

```json {class="line-numbers linkable-line-numbers"}
{
  "attributes": {
    "pins": [
      <int>
    ],
    "analogs": [
      {

>         "pin": "<number>",

Reviewer 2: ```suggestion
        "pin": "<GPIO-number>",
```
For more clarity esp. since this is different from other boards

Author: Ahh good suggestion but I realized this is actually an int like the pin number above! @gvaradarajan can you confirm this is not a string?

>         "pin": "<number>",

Author: ```suggestion
        "pin": <int>,
```

        "name": "<your-analog-name>"
      }
    ]
  },
  "depends_on": [],
  "model": "esp32",
  "name": "<your-board-name>",

>   "type": "board"

Reviewer 2: Move to top of config and add namespace if applicable. I've been ordering everything name, model, type, namespace, attributes, depends_on for consistency, and also because that's how the app seems to autopopulate it these days, and also because I think name model and type make a lot more sense at the beginning, to sort of introduce the component, before getting into the details.

Author: That makes a lot of sense, done! TY!

}
```

{{% /tab %}}
{{% tab name="JSON Example" %}}

```json {class="line-numbers linkable-line-numbers"}
{
  "attributes": {
    "pins": [15],
    "analogs": [
      {
        "pin": "34",
        "name": "sensor"
      }
    ]
  },
  "depends_on": [],

>   "model": "esp32",

Reviewer 2: See comment above

Author: Done!

  "name": "board",
  "type": "board"
}
```

{{% /tab %}}
{{< /tabs >}}

Edit and fill in the attributes as applicable.
Click **Save config**.

The following attributes are available for `esp32` boards:

<!-- prettier-ignore -->
| Name | Type | Inclusion | Description |
| ---- | ---- | --------- | ----------- |
| `analogs` | object | Optional | Attributes of any pins that can be used as analog-to-digital converter (ADC) inputs. See [configuration info](#analogs). |
| `i2cs` | object | Optional | Any Inter-Integrated Circuit (I<sup>2</sup>C) pins' bus index and name. See [configuration info](#i2cs). |
| `digital_interrupts` | object | Optional | Any digital interrupts' GPIO number. See [configuration info](#digital_interrupts). |
| `pins` | object | Required | The GPIO number of any GPIO pins you wish to use as input/output with the [`GPIOPin` API](/program/apis/#gpio-pins). |

> Any pin not specified in either `"pins"` or `"digital_interrupts"` cannot be interacted with through the [Board API](/components/board/#api).

Reviewer 2: ```suggestion
Any pin not specified in either `"pins"` or `"digital_interrupts"` cannot be interacted with through the [board API](/components/board/#api).
```

Author: TY, done

> Interaction with digital interrupts is only supported with the [Board API](/components/board/#api), these digital interrupts cannot be used as software interrupts in driver implementations.

Reviewer 2: ```suggestion
Interaction with digital interrupts is only supported with the [board API](/components/board/#api); these digital interrupts cannot be used as software interrupts in driver implementations.
```

Author: Done!

> {{< alert title="Info" color="info" >}}

Reviewer 2: Consider moving this tip up above the JSON template/example tabs, since that's where users may be used to finding a builder mode tab

Reviewer: +1

Author: moved!

> The`esp32` [board](/components/board/) model is not currently provided for you as a built-in option in [the Viam app](https://app.viam.com), so you cannot use the **Config Builder** to configure this board.

Reviewer 2: ```suggestion
The`esp32` [board](/components/board/) model is not currently available as a built-in option in [the Viam app](https://app.viam.com), so you cannot use **Builder** mode to configure this board.
```
To better match the UI labels

Author: Done!

---
title: "Configure an ESP32 board"
linkTitle: "esp32"
weight: 20
type: "docs"
description: "Configure an esp32 board."
images: ["/icons/components/board.svg"]
tags: ["board", "components"]
# SMEs: Gautham, Rand
---

{{% alert title="REQUIREMENTS" color="caution" %}}

Follow the [setup guide](/installation/prepare/microcontrollers/) to prepare your ESP32 for running the micro-RDK before configuring an `esp32` board.

{{% /alert %}}

> 

Author: ```suggestion

{{% readfile "/static/include/micro-rdk-hardware.md" %}}

```

---
title: "Configure an ESP32 board"
linkTitle: "esp32"
weight: 20
type: "docs"
description: "Configure an esp32 board."
images: ["/icons/components/board.svg"]
tags: ["board", "components"]
# SMEs: Gautham, Rand
---

{{% alert title="REQUIREMENTS" color="caution" %}}

Follow the [setup guide](/installation/prepare/microcontrollers/) to prepare your ESP32 for running the micro-RDK before configuring an `esp32` board.

{{% /alert %}}

> {{% readfile "/static/include/micro-rdk-hardware.md" %}}

Author: @randhid added that hardware info here!

---
title: "Configure an ESP32 board"
linkTitle: "esp32"
weight: 20
type: "docs"
description: "Configure an esp32 board."
images: ["/icons/components/board.svg"]
tags: ["board", "components"]
# SMEs: Gautham, Rand
---

{{% alert title="REQUIREMENTS" color="caution" %}}

Follow the [setup guide](/installation/prepare/microcontrollers/) to prepare your ESP32 for running the micro-RDK before configuring an `esp32` board.

Viam recommends purchasing the ESP32 with a development board. The following ESP32 microcontrollers are supported:

> 

Author: @randhid added hardware info here re. request, this seem ok?
